internal/model/model.go

```diff
@@ -1129,19 +1132,33 @@ func (m *Model) ScanFolderSub(folder, sub string) error {
 
 	// Required to make sure that we start indexing at a directory we're already
 	// aware off.
-	for sub != "" {
-		if _, ok = fs.Get(protocol.LocalDeviceID, sub); ok {
-			break
+	var unifySubs []string
+	for _, sub := range subs {
+		for sub != "" {
+			if _, ok = fs.Get(protocol.LocalDeviceID, sub); ok {
+				break
+			}
+			sub = filepath.Dir(sub)
+			if sub == "." || sub == string(filepath.Separator) {
+				sub = ""
+			}
 		}
-		sub = filepath.Dir(sub)
-		if sub == "." || sub == string(filepath.Separator) {
-			sub = ""
+		seenPrefix := false
+		for _, us := range unifySubs {
+			if strings.HasPrefix(sub, us) {
+				seenPrefix = true
+				break
```

Comment:
This could be simplified by adding a label `nextSub:` before the outer `for` and then just `continue nextSub` here instead of the `seenPrefix`+`break`+`if`.

We might still do unnecessary work if we first add `foo/bar/baz` and then `foo/bar` in the next loop, but perhaps that's no big deal? If it is, just adding them all and then doing a sort on the results and using a `seen map[string]bool` afterwards might help...